airflow/providers/apache/spark/hooks/spark_submit.py

```diff
@@ -421,11 +421,18 @@ def submit(self, application="", **kwargs):
         # Check spark-submit return code. In Kubernetes mode, also check the value
         # of exit code in the log, as it may differ.
         if returncode or (self._is_kubernetes and self._spark_exit_code != 0):
-            raise AirflowException(
-                "Cannot execute: {}. Error code is: {}.".format(
-                    self._mask_cmd(spark_submit_cmd), returncode
+            if self._is_kubernetes:
+                raise AirflowException(
+                    "Cannot execute: {}. Error code is: {}.".format(
+                        self._mask_cmd(spark_submit_cmd), returncode
+                    )
+                )
+            else:
+                raise AirflowException(
+                    "Cannot execute: {}. Error code is: {}. Kubernetes spark exit code is: {}".format(
```

Comment:
Why does this say "Kubernetes Spark". This is in the else so this is not Spark on K8s, isn't it?

Reply:
Thanks for catching that will update the PR